src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1516,6 +1525,291 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
         raise CLIInternalError("Exception caught while running troubleshoot: {}".format(str(ex)))
 
 
+def display_diagnostics_report(temp_kubeconfig_path, kube_config):   # pylint: disable=too-many-statements
+    if not which('kubectl'):
+        raise CLIError('Can not find kubectl executable in PATH')
+    subprocess_cmd = ["kubectl", "get", "node", "--no-headers"]
+    if kube_config:
+        subprocess_cmd.extend(["--kubeconfig", kube_config])
+    nodes = subprocess.check_output(
+        subprocess_cmd,
+        universal_newlines=True)
+    logger.debug(nodes)
+    node_lines = nodes.splitlines()
+    ready_nodes = {}
+    for node_line in node_lines:
+        columns = node_line.split()
+        logger.debug(node_line)
+        if columns[1] != "Ready":
+            logger.warning("Node %s is not Ready. Current state is: %s.", columns[0], columns[1])
+        else:
+            ready_nodes[columns[0]] = False
+
+    logger.debug('There are %s ready nodes in the cluster', str(len(ready_nodes)))
+
+    if not ready_nodes:
+        logger.warning('No nodes are ready in the current cluster. Diagnostics info might not be available.')
+
+    network_config_array = []
+    network_status_array = []
+    apds_created = False
+
+    max_retry = 10
+    for retry in range(0, max_retry):
+        if not apds_created:
+            subprocess_cmd = ["kubectl", "get", "apd", "-n", "aks-periscope", "--no-headers"]
```

Comment:
E.g, here you will need just subprocess_cmd = kubectl_prior + ["get", "apd ......]